src/connectors/CustomNetworkConnector.ts

```diff
@@ -126,12 +126,13 @@ export class CustomNetworkConnector extends AbstractConnector {
   public changeChainId(chainId: number) {
     invariant(Object.keys(this.providers).includes(chainId.toString()), `No url found for chainId ${chainId}`)
     this.currentChainId = chainId
-    this.emitUpdate({ provider: this.providers[this.currentChainId], chainId })
+    return new Promise(() => this.emitUpdate({ provider: this.providers[this.currentChainId], chainId }))
```

Comment:
why do you want to return it as `promise`?

Reply:
I want a promise to have every function return a promise. Maybe I can do when I set `changeChainId` in `useNetworkSwitch`